src/blockdata/block.rs

```diff
@@ -141,9 +141,11 @@ impl BlockHeader {
             return Err(BlockBadTarget);
         }
         let block_hash = self.block_hash();
-        let mut ret = [0u64; 4];
-        util::endian::bytes_to_u64_slice_le(block_hash.as_inner(), &mut ret);
-        let hash = &Uint256(ret);
+
+        let mut bytes = block_hash.into_inner();
+        bytes.reverse();
+
+        let hash = &Uint256::from_be_bytes(bytes);
```

Comment:
Why a reference?

Reply:
Looks like it isn't needed indeed. I guess some chain of refactors caused it.

Reply:
I was getting an unknown size compile error without the reference. I won't bother investigating because this is modified by #1197

Reply:
Did you remove reference from *both* sides?

Reply:
I'm a goose, I must have been mixed up. There is no compiler warning.